test_unit.py

```diff
@@ -1788,6 +1829,323 @@ def test_transaction(self):
+        self.assertEqual(actual.verify(), expectedValid)
+
+        if not expectedValid:
+            return
+
+        actualEncoded = encoding.msgpack_encode(actual)
+        self.assertEqual(actualEncoded, expectedEncoded)
+
+        decoded = encoding.future_msgpack_decode(actualEncoded)
+        self.assertEqual(decoded, actual)
+
+    def test_LogicSig_escrow(self):
+        lsig = transaction.LogicSig(TestLogicSigTransaction.program, TestLogicSigTransaction.args)
+
+        sender = lsig.address()
+        expected = "gqRsc2lngqNhcmeSxAEBxAICA6FsxAUBIAEBIqN0eG6Ko2FtdM0TiKNmZWXOAANPqKJmds4ADtbco2dlbq10ZXN0bmV0LXYzMS4womdoxCAmCyAJoJOohot5WHIvpeVG7eftF+TYXEx4r7BFJpDt0qJsds4ADtrEpG5vdGXECLRReTn8+tJxo3JjdsQgtMYiaKTDNVD1im3UuMojnJ8dELNBqn4aNuPOYfv8+Yqjc25kxCD2di2sdbGZfWwslhgGgFB0kNeVES/+f7dgsnOK+cfxraR0eXBlo3BheQ=="
+        self._test_sign_txn(lsig, sender, expected)
+    
+    def test_LogicSig_escrow_different_sender(self):
+        lsig = transaction.LogicSig(TestLogicSigTransaction.program, TestLogicSigTransaction.args)
+
+        sender = TestLogicSigTransaction.otherAddr
+        expected = "g6Rsc2lngqNhcmeSxAEBxAICA6FsxAUBIAEBIqRzZ25yxCD2di2sdbGZfWwslhgGgFB0kNeVES/+f7dgsnOK+cfxraN0eG6Ko2FtdM0TiKNmZWXOAANPqKJmds4ADtbco2dlbq10ZXN0bmV0LXYzMS4womdoxCAmCyAJoJOohot5WHIvpeVG7eftF+TYXEx4r7BFJpDt0qJsds4ADtrEpG5vdGXECLRReTn8+tJxo3JjdsQgtMYiaKTDNVD1im3UuMojnJ8dELNBqn4aNuPOYfv8+Yqjc25kxCC0xiJopMM1UPWKbdS4yiOcnx0Qs0Gqfho2485h+/z5iqR0eXBlo3BheQ=="
+        self._test_sign_txn(lsig, sender, expected)
+    
+    def test_LogicSig_single_delegated(self):
+        sk = mnemonic.to_private_key("olympic cricket tower model share zone grid twist sponsor avoid eight apology patient party success claim famous rapid donor pledge bomb mystery security ability often")
+
+        lsig = transaction.LogicSig(TestLogicSigTransaction.program, TestLogicSigTransaction.args)
+        lsig.sign(sk)
+
+        sender = account.address_from_private_key(sk)
+        expected = "gqRsc2lng6NhcmeSxAEBxAICA6FsxAUBIAEBIqNzaWfEQD4FPTlN+xK8ZXmf6jGKe46iUYtVLIq+bNenZS3YsBh+IQUtuSRiiRblYXTNDxmsuWxFpCmRmREd5Hzk/BLszgKjdHhuiqNhbXTNE4ijZmVlzgADT6iiZnbOAA7W3KNnZW6tdGVzdG5ldC12MzEuMKJnaMQgJgsgCaCTqIaLeVhyL6XlRu3n7Rfk2FxMeK+wRSaQ7dKibHbOAA7axKRub3RlxAi0UXk5/PrScaNyY3bEILTGImikwzVQ9Ypt1LjKI5yfHRCzQap+GjbjzmH7/PmKo3NuZMQgXmdPHAru7DdxiY9hx2/10koZeT4skfoIUWJj44Vz6kKkdHlwZaNwYXk="
+        self._test_sign_txn(lsig, sender, expected)
+    
+    def test_LogicSig_single_delegated_different_sender(self):
+        sk = mnemonic.to_private_key("olympic cricket tower model share zone grid twist sponsor avoid eight apology patient party success claim famous rapid donor pledge bomb mystery security ability often")
+
+        lsig = transaction.LogicSig(TestLogicSigTransaction.program, TestLogicSigTransaction.args)
+        lsig.sign(sk)
+
+        sender = TestLogicSigTransaction.otherAddr
+        self._test_sign_txn(lsig, sender, None, False)
```

Comment:
Q: Why is this tx rejected when the test_LogicSigAccount_single_delegated_different_sender() tx is approved?

Reply:
This case is impossible to support, since the `LogicSig` object doesn't know the public key that signed it. That's why `LogicSigAccount` had to be introduced.

The test is just to make sure it doesn't succeed.